lightningd/peer_control.c

```diff
@@ -176,6 +176,57 @@ u8 *p2wpkh_for_keyidx(const tal_t *ctx, struct lightningd *ld, u64 keyidx)
 	return scriptpubkey_p2wpkh(ctx, &shutdownkey);
 }
 
+static struct simple_htlc **collect_htlcs(struct channel *channel, u32 local_feerate) {
+	// Collect the htlcs for call to hsmd.
+	struct htlc_in_map *htlcs_in = &channel->peer->ld->htlcs_in;
+	struct htlc_out_map *htlcs_out = &channel->peer->ld->htlcs_out;
+	struct simple_htlc **htlcs = tal_arr(tmpctx, struct simple_htlc *, 0);
+
+	const struct htlc_in *hin;
+	struct htlc_in_map_iter ini;
+	for (hin = htlc_in_map_first(htlcs_in, &ini);
+	     hin;
+	     hin = htlc_in_map_next(htlcs_in, &ini)) {
+		if (hin->key.channel != channel)
+			continue;
+		if (htlc_is_trimmed(REMOTE, hin->msat, local_feerate,
+				    channel->our_config.dust_limit, LOCAL,
+				    channel_has(channel, OPT_ANCHOR_OUTPUTS)))
+			continue;
+		struct simple_htlc *simple =
+		    new_simple_htlc(NULL,
+				    htlc_state_owner(hin->hstate),
+				    hin->msat,
+				    &hin->payment_hash,
+				    hin->cltv_expiry
+				    );
```

Comment:
I think code convention for this repo has this on the same line as the last arg

Reply:
fixed